### DISCUSSION TITLE
Best practice for working with unsupported types?

### DISCUSSION THREAD

**Discussion by Xetera** [prisma/prisma#4790] [Q&A] [answered]
That's the right approach for using prisma with unsupported types? I need the `CUBE` extension on postgres to do face recognition but because this extension isn't supported by prisma yet, it comments out the field in my schema which breaks the insert queries I'm doing.

Do you normally do a raw SQL query after the prisma one to insert rows that include an unsupported row? Do you insert as much as you can with prisma and then update newly inserted rows with the unsupported data type through raw queries?

**Comment by nikolasburk** [comment, marked as answer]
Hey @Xetera, this is a great question and your hunch is already correct! If a type is not yet natively supported by Prisma, the best way to send read/write queries for that columns of that type is by using [raw SQL](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) queries. 

> Do you normally do a raw SQL query after the prisma one to insert rows that include an unsupported row? Do you insert as much as you can with prisma and then update newly inserted rows with the unsupported data type through raw queries?

I don't think there's _the_ way to do it, but how you structure your queries always depends on what exact data you need to write into the DB. If you can give an example, I can try to help figuring out the best approach for you!

Also, if you'd like to see native support for the `CUBE` extension in Prisma, it would be great if you could [open a feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this and describe your use case. That way our product and engineering teams can look into it! 

Please let me know if that helps or if you have any further questions, always happy to help 🙌

**Reply by JakubKoralewski** [reply]
I'm also interested in this. I'm looking in the Prisma docs, but I don't see a clear way that's described for dummies like me to understand. 

Let's say I want to have the postgres `daterange` field on one of my models. Do I annotate the field as a string in the `.prisma` schema, then modify the migration SQL script and replace `TEXT` with `daterange`? Then also modify the generated type by doing something like ``const returnValue: Omit<NormalReturnType, 'customField'> & {customField: ????} = await prisma.$queryRaw`SELECT * FROM table_with_the_custom_field` `` ?

A description for this as one of the [Prisma Migrate Flows](https://www.prisma.io/docs/concepts/components/prisma-migrate/prisma-migrate-flows) would be nice. Unless I'm missing it.

**Reply by dukedorje** [reply]
Unsupported("daterange")

**Reply by budyclub** [reply]
Select would not work 😒

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
